datafusion/datasource-parquet/src/opener.rs

```diff
@@ -178,7 +182,7 @@ impl FileOpener for ParquetOpener {
             // Build predicates for this specific file
             let (pruning_predicate, page_pruning_predicate) = build_pruning_predicates(
                 predicate.as_ref(),
-                &physical_file_schema,
+                &logical_file_schema,
```

Comment:
This is the actual change (from physical to logical schema) -- I am calling this out because it took me a while to spot it (at first I thought this was only a name change)

Reply:
Yes sorry for also changing the names in this PR - it was just too confusing to even work on this PR without giving things a new name. Let me know if you have any suggestions for better names, happy to accept them, naming here seems to be 1/2 of the problem. You know what they say, there's 3 hard problems...

Reply:
no worries -- I think the new names are much clearer to me

Reply:
Sorry, coming to this too late, but I'm curious why the logical schema? To me pruning implies a physical layer kind of thing. I ask because now I'm having issues in #15821 trying to build pruning predicates for columns that don't exist in the physical schema.

This may just mean #15821 is operating at too low a level and needs to operate up here where we a) know it's parquet and b) have the physical schema.

@alamb @adriangb

Reply:
I think the key observation is that the predicates are expressed in terms of the table schema (not the file schema). 

Today there is a bunch of logic to translate table schema -> file schema  for data and statistics.

So in order to evaluate predicates on the physical schema we would need to translate the expression as well, which I think is what @adriangb  is saying above

Reply:
Yep precisely. And that requires re evaluating cast rules, etc which is... not easy.

Reply:
Ok, thanks for the explanation!